### DISCUSSION TITLE
Add users to couch DB using `local.ini`

### DISCUSSION THREAD

**Discussion by ismailmmd** [apache/couchdb#3749] [Q&A] [answered]
How to add non admin users to CouchDB using ```local.ini``` configuration file?

**Comment by eiri** [comment, marked as answer]
You can't. Only admin users can be added in `local.ini`, that's why the according section called `[admins]`.

Non admin users created by adding user documents into `_users` database as described in [1.5.2. Authentication Database](https://docs.couchdb.org/en/stable/intro/security.html#authentication-database)

**Reply by ismailmmd** [reply]
Thank you. One more quick question, the doc says -
`
roles (array of string): List of user roles. CouchDB doesn’t provide any built-in roles, so you’re free to define your own depending on your needs. However, you cannot set system roles like _admin there. Also, only administrators may assign roles to users - by default all users have no roles
`
Is there a possible way to add users with specific privileges.

**Reply by eiri** [reply]
Yes, to some extend. I wrote a post on the topic a couple of years ago https://medium.com/@eiri/couchdb-authorization-in-a-database-58c8ee633c96. Some details maybe outdated, but core concepts should hold. I hope you'll find it useful.
